<a href="https://colab.research.google.com/github/Phantom6Paradise7/Data_Science_2026/blob/main/Data_science_Day0_%2B_Day1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
print("Hello World!")

Hello World!


In [ ]:
print("Hello Machines...")

Hello Machines...


In [ ]:
a = 10
print(a * 2 + 5)

25


In [ ]:
a = 5
b = 10
c = 15
d = 20
e = 25
print(a+b+c+d+e)

75


In [ ]:
a = input()
b = input()
c = input()
print(a+b+c)

1 2 3 
2 3 4
5 6 8
1 2 3 2 3 45 6 8


In [ ]:
a = input()
b = input()
c = input()
print(a+b+c)

1
2
3
123


In [ ]:
a = 1 * 10**9
b = 1 * 10**8
print(a+b)

1100000000


In [ ]:
a = 1
b = 5
# print(a + b + c)

In [ ]:
a = 10
b = 20
a, b = b , a
print (a, b)

20 10


In [ ]:
for i in range(1, 51):
    if i % 2 == 0:
        print("Even no :", i)
    else:
        print("Odd no :", i)


Odd no : 1
Even no : 2
Odd no : 3
Even no : 4
Odd no : 5
Even no : 6
Odd no : 7
Even no : 8
Odd no : 9
Even no : 10
Odd no : 11
Even no : 12
Odd no : 13
Even no : 14
Odd no : 15
Even no : 16
Odd no : 17
Even no : 18
Odd no : 19
Even no : 20
Odd no : 21
Even no : 22
Odd no : 23
Even no : 24
Odd no : 25
Even no : 26
Odd no : 27
Even no : 28
Odd no : 29
Even no : 30
Odd no : 31
Even no : 32
Odd no : 33
Even no : 34
Odd no : 35
Even no : 36
Odd no : 37
Even no : 38
Odd no : 39
Even no : 40
Odd no : 41
Even no : 42
Odd no : 43
Even no : 44
Odd no : 45
Even no : 46
Odd no : 47
Even no : 48
Odd no : 49
Even no : 50


In [ ]:
a = 2
if a % 2 == 0:
  print("Even")
else :
  print("Odd")

Even


In [ ]:
a = 0
if a > 0:
  print("Positive")
elif a == 0:
  print("Zero")
else :
  print("Negative")

Zero


In [ ]:
a = input()
if a == "Sun":
  print("Sunday")
elif a == "Mon":
  print("Monday")
elif a == "Tue":
  print("Tuesday")
else :
  print("Wednesday")

tue
Wednesday


In [ ]:
for i in range(0, 11, 2):
  print(i)

0
2
4
6
8
10


In [ ]:
for i in range(1, 11, 2):
  print(i)

1
3
5
7
9


In [ ]:
a = 0
while a < 10:
  print(a)
  a = a+1

0
1
2
3
4
5
6
7
8
9


In [ ]:
s = input()
rev = ""
for ch in s:
  rev = ch + rev
if s == rev:
  print("Pallindrome")
else :
  print("Not A Pallindrome")

HomeemoH
Pallindrome


In [ ]:
lst = [1, 2, 3, "Hi", 'Hi', 'true', 'false', 'c', [2,3]]
print(
    lst
)

[1, 2, 3, 'Hi', 'Hi', 'true', 'false', 'c', [2, 3]]


Array can only store same type of datatype

While List can store heterogeneous type of data

we fetch array from numpy

and list takes time to execute

In [ ]:
lo = [182, 81, 'yo', 'yu', [8812, [199, 188], 881]]
lo.append(2)
print(lo)
lo.append([82, [818, 716,1445]])
print(lo)
lo.extend([2, [881, [7612, [9182, 61552]]], 7612, 'true'])
print(lo)
lo.insert(2, 'false')
print(lo)

#append - adds element at last - every value
#extend - to add multiple elements in our list, requires an iterable data structure (at last), and seperates all the elements
#insert - to add element at a custom value of index - in between



[182, 81, 'yo', 'yu', [8812, [199, 188], 881], 2]
[182, 81, 'yo', 'yu', [8812, [199, 188], 881], 2, [82, [818, 716, 1445]]]
[182, 81, 'yo', 'yu', [8812, [199, 188], 881], 2, [82, [818, 716, 1445]], 2, [881, [7612, [9182, 61552]]], 7612, 'true']
[182, 81, 'false', 'yo', 'yu', [8812, [199, 188], 881], 2, [82, [818, 716, 1445]], 2, [881, [7612, [9182, 61552]]], 7612, 'true']


In [ ]:
ok = [20]
ok.append('hi')
ok.extend(['everyone', 'how is life going'])
ok.insert(1, 'Good Morning and')
print(ok)

[20, 'Good Morning and', 'hi', 'everyone', 'how is life going']


In [ ]:
# Remove function from a list - pop
#pop - remove from the end
#remove - requires value not index and returns null
#index - returns index of the value
#count - tells how many times an element is in an index
#sort - sorts the list (lexographically in string case) - preference of homogenous list
lo = [182, 81, 'yo', 'yu', [8812, [199, 188], 881], 91, 81, 21]
lo.pop()
print(lo)
lo.remove(182)
print(lo)
print(lo.index(81))
print(lo.count(81))
op = [9191, 7199,6457,123456,987654,7623,2,18731,981342]
op.sort()
print(op)

[182, 81, 'yo', 'yu', [8812, [199, 188], 881], 91, 81]
[81, 'yo', 'yu', [8812, [199, 188], 881], 91, 81]
0
2
[2, 6457, 7199, 7623, 9191, 18731, 123456, 981342, 987654]


In [ ]:
lol = [12, 34, 56, 78, 910, -109, -87, -65, -43, -21, 0]
print(sum(lol))
pole = [6, 78, 910, -109, -8]
print(min(pole))
print(max(pole))
print(sorted(pole))
#sort - is a method
#sorted - is a function in which list goes as a parameter
#sum - returns sum of the list
#min - smallest element return
#max - largest element return

765
-109
910
[-109, -8, 6, 78, 910]


In [ ]:
lst = [-66, 56, 42, 907, 235]
a = sorted(lst)
print(a)

[-66, 42, 56, 235, 907]


In [ ]:
#lists that are immutable
tp = (21,93,56,74,80)
print(tp)
u = sorted(tp)
print(u)
#sorted function returns a list that are changable (mutable) hence the answer we get on using sorted function on a tuple is a list

(21, 93, 56, 74, 80)
[21, 56, 74, 80, 93]


Tuple Functions

count values - count()

find - index()

size - len()

total - sum()

highest/lowest - max()/min()

sort - sorted() [returns list]

In [ ]:
Tup = (20,43,95,76,81,20)
print(Tup.count(20)) #count - used to tell how many times a number returns
print(Tup.index(95)) #index - used to find the index of a value provided
print(len(Tup)) #len - tells the length of the tuple
print(sum(Tup)) #sum - sum of all the elements in the tuple
print(max(Tup)) #max - returns the maximum element
print(min(Tup)) #min - returns the maximum element
print(sorted(Tup)) #sorted - returns list after sorting a tuple

2
2
6
335
95
20
[20, 20, 43, 76, 81, 95]


In [ ]:
l1 = []
l2 = []
size = int(input()) #size is to be given in one digits - don't forget
for i in range(size):
  x = int(input())
  l1.append(x)
for i in range(size):
  y = int(input())
  l2.append(y)
l3 = []
for i in range(size):
  l3.append(l1[i] + l2[i])
print(l3)
  # l3[i] = l1[i] + l2[i] - index error since l3 meine koi element hai hi nhi
  # print(l3)

3
2
7
9
2
6
4
[4, 13, 13]


# input should be given in different line like - 1
#                                           2
  #                                             3
not like 123
or 1 2 3